index.js

```diff
@@ -3,6 +3,46 @@
 module.exports = {
   name: require('./package').name,
 
+  options: {
+    babel: {
+      plugins: [
+        // Ensure that `ember-auto-import` can handle the dynamic imports
+        require('ember-auto-import/babel-plugin')
+      ]
+    }
+  },
+
+  addonJsFiles() {
+    const Funnel = require('broccoli-funnel');
+    const config = this._findHost().options[this.name];
+
+    const tree = this._super.addonJsFiles.apply(this, arguments);
+
+    if (config && config.lazyLoad === true) {
```

Comment:
Can this be namespaced under the other mapbox-gl configurations? Maybe I'm misunderstanding.

`ENV[`mapbox-gl`].lazyLoad = true`

Reply:
currently the option must come from the ember app config as seen [here](https://github.com/kturney/ember-mapbox-gl/pull/67/files#diff-2fe2a7faee7de1e0d5d82e032498d84cR7)

It uses `this.name` for option namespacing.

Reply:
Hey there,
Thanks for the heads up. So, I have come up with a possible solution to handle lazy loading transparently on engines.
Following code replaces rows 17 to 21:
```javascript
    let app;
    let current = this;
    let lazyEngine = false;
    do {
	    if (current.options.lazyLoading && current.options.lazyLoading.enabled)
	    	lazyEngine = current.options.lazyLoading.enabled;
      app = current.app || app;
    } while (current.parent.parent && (current = current.parent));

		const config = app.options[this.name];


    const tree = this._super.addonJsFiles.apply(this, arguments);

    if ((config && config.lazyLoad === true) || lazyEngine) {
```
What I am doing is basically running the same `do... while` loop that `_findHost` does, but always overwriting the `lazyEngine` variable with whatever is written inside the `options.lazyLoading.enabled` property. This is the same property that is set by default when setting an engine as lazy in ember-engines. This way, `lazyEngine` will have the same value as the topmost-engine `lazyLoading` option.
I haven't tested it in a standalone app, but works great as an in-engine add-on.
Let me know what you think!